### PR TITLE
`linalg_lu_factor_ex`: change to new `MetaBase` API.

### PR DESCRIPTION
**Pull request by ysiraichi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #78039
* #78038
* #77540
* __->__ #77539
* #76098
* #76097

This PR modifies `linalg_lu_factor_ex` to the new `MetaBase` API. Instead of checking the
output strides manually, it now uses the appropriate `set_output_strided` and
`set_output_contiguous`.

Add for lu

### CODE REVIEW

File: test/test_linalg.py [hunk 1]
```diff
@@ -5211,6 +5211,39 @@ def test_householder_product_errors_and_warnings(self, device):
...
+
+        fn(A, out=out_)
+
+        for i in range(len(out)):
+            # Check whether the results are equal for the 'out' overload.
+            self.assertEqual(out[i], out_[i])
+            # Check whether the strides were preserved after the call.
+            self.assertEqual(strides_[i], out_[i].stride())
+            # Check whether the original strides are different from the generated ones.
+            self.assertNotEqual(out[i].stride(), out_[i].stride())
```
Comment (Lezcano) [inline test/test_linalg.py:5234]: I believe these things are already tested in the `OpInfo` tests in
https://github.com/pytorch/pytorch/blob/c83f8ee46ab057a8bce7ed7ba7ceccba5c388d8f/test/test_ops.py#L676-L684

**Review by lezcano — COMMENTED** [review]
This is fantastic! Thank you so much!

The only thing here is that I believe that the whole test is not necessary, as these things are already tested in the `OpInfo` for this operation.

File: test/test_linalg.py (hunk 1 above)
Comment (ysiraichi) [reply test/test_linalg.py:5234]: Ah, you are right! Thanks for pointing that out.

**Review by lezcano — APPROVED** [review]
This is just great :D

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**